django/db/models/fields/__init__.py

```diff
@@ -921,6 +921,8 @@ def get_db_prep_value(self, value, connection, prepared=False):
 
     def get_db_prep_save(self, value, connection):
         """Return field's value prepared for saving into a database."""
+        if hasattr(value, "as_sql"):
```

Comment:
Is this to support passing through `Value`?

Reply:
Right, that to account for the fact that in the third commit both `SQLUpdateCompiler` and `SQLInsertCompiler` now pass compilable but the pre-existing default implementations of `get_db_prep` is not compilable proof by default. That's the change I was referring to in https://github.com/django/django/pull/16245#discussion_r1014762459